module/module_test.go

```diff
@@ -248,3 +346,345 @@ func TestModuleFunctions(t *testing.T) {
 	err = myRobot.Close(ctx)
 	test.That(t, err, test.ShouldBeNil)
 }
+
+// mockSummation mocks the summation service so we can ensure modular addition
+// and reconfiguration correctly set up resources without a Reconfigure method.
+type mockSummation struct {
+	deps   registry.Dependencies
+	config *MockSummationConfig
+}
+
+type MockSummationConfig struct {
+	Motors []string `json:"motors"`
+}
+
+func (c *MockSummationConfig) Validate(path string) ([]string, error) {
+	if len(c.Motors) < 1 {
+		return nil, errors.New("required attributes 'motors' not specified or empty")
+	}
+	return c.Motors, nil
+}
+
+func (mock *mockSummation) Sum(ctx context.Context, nums []float64) (float64, error) {
+	return 0, errors.New("sum unimplemented for mockSummation")
+}
+
+// mockSummationWithReconfigure mocks the summation service so we can ensure
+// modular addition and reconfiguration correctly set up resources with a Reconfigure
+// method.
+type mockSummationWithReconfigure struct {
+	mockSummation
+	reconfigureCallCount int
+}
+
+func (mock *mockSummationWithReconfigure) Reconfigure(ctx context.Context,
+	cfg config.Service, deps registry.Dependencies,
+) error {
+	mock.reconfigureCallCount++
+	conf, ok := cfg.ConvertedAttributes.(*MockSummationConfig)
+	if !ok {
+		return rdkutils.NewUnexpectedTypeError(conf, cfg.ConvertedAttributes)
+	}
+	mock.config = conf
+	mock.deps = deps
+	return nil
+}
+
+// TestModularResourceWithoutReconfigure tests that modularly-handled resources
+// without a Reconfigure method are correctly set up when initially added and
+// later reconfigured.
+func TestModularResourceWithoutReconfigure(t *testing.T) {
```

Comment:
I tried to combine `TestModularResourceWithoutReconfigure` and `TestModularResourceWithReconfigure` into a single test to reduce code duplication. While the code looks similar, there are very subtle differences between the tests that make them hard to combine. In general, I'm ok with duplication in testing code that allows for easier readability. Combining the two makes these already complex tests very difficult to read IMO.